### PR TITLE
preflight-checks: Disallow `root` `ansible_user`

### PR DESCRIPTION
**Pull request by NicolasT**
Fixes: #329
See: https://github.com/scality/metalk8s/issues/329

### CODE REVIEW

**Comment by NicolasT** [comment]
@bert-e wait

At least until @wabernatScality approves.

**Review by Zempashi — DISMISSED** [review]

**Comment by NicolasT** [comment]
Noooo. Why is this in the queue? There's a 'wait' comment... @tcarmet

**Comment by NicolasT** [comment]
Was: @bert-e wait

**Comment by NicolasT** [comment]
Had to retarget base branch btw

File: ChangeLog.rst [hunk 1]
```diff
@@ -10,6 +10,8 @@ Features added
 
 :ghpull:`305` - ensure that journald logs are persisted across reboot (:ghissue:`303`)
 
+:ghpull:`337` - assert `ansible_user` is not `root` (:ghissue:`329`)
```
Comment (lucieleonard) [inline ChangeLog.rst]: ensure instead of assert

File: docs/faq.rst [hunk 2]
```diff
@@ -49,3 +49,27 @@ There are two ways to configure Ansible to keep a logfile:
   :command:`ansible-playbook` will be invoked.
 
 For more information, see :ref:`ansible:DEFAULT_LOG_PATH`.
+
+
+.. _ansible-user-root-detection:
+
+Can I use the 'root' user to deploy MetalK8s to servers?
+--------------------------------------------------------
+During deployment of MetalK8s, a set of tasks are executed to bring the target
```
Comment (lucieleonard) [inline docs/faq.rst]: During the deployment

File: docs/faq.rst [hunk 3]
```diff
@@ -49,3 +49,27 @@ There are two ways to configure Ansible to keep a logfile:
...
+system in line with the `RHEL7 STIG`_ security guidelines, using the
+`ansible-hardening`_ role. STIG rule `V-72247`_ disallows remote SSH access
```
Comment (lucieleonard) [inline docs/faq.rst]: disables remote SSH access

File: docs/faq.rst [hunk 4]
```diff
@@ -49,3 +49,27 @@ There are two ways to configure Ansible to keep a logfile:
...
+using the `root` user. As such, if MetalK8s were deployed using `root` to
+access a remote system, this would effectively disable access to said server.
+
+We integrated a check in the playbook to assert `ansible_user` is not set to
+`root` on any of the target hosts, so deployment will abort early if this
```
Comment (lucieleonard) [inline docs/faq.rst]: to abort any deployment if this configuration is detected.

File: docs/faq.rst [hunk 5]
```diff
@@ -49,3 +49,27 @@ There are two ways to configure Ansible to keep a logfile:
...
+configuration is detected.
+
+If you'd rather disable this security measure, set the
```
Comment (lucieleonard) [inline docs/faq.rst]: To disable this feature, ...

**Review by lucieleonard — CHANGES_REQUESTED** [review]
Wording and/or grammar changes

File: ChangeLog.rst (hunk 1 above)
Comment (NicolasT) [reply ChangeLog.rst]: As discussed offline: this is the `ChangeLog`. The change is we now `assert` on the user being used. We don't (or even more, can't) *ensure* it's not `root` by taking whatever action but failing.

File: docs/faq.rst (hunk 3 above)
Comment (NicolasT) [reply docs/faq.rst]: Will change to `does not permit` after offline chat.

File: docs/faq.rst (hunk 4 above)
Comment (NicolasT) [reply docs/faq.rst]: Done,  but replaced `any` with `the` as discussed offline.

**Review by lucieleonard — APPROVED** [review]

**Pull request merged**